docs - multi-catalog support for databricks (#58901)
metabase/metabase: docs/databases/connections/databricks.md

See [Authenticate access with a service principal using OAuth](https://docs.databricks.com/en/dev-tools/auth/oauth-m2m.html).

### Enable multiple catalogs

> Toggle on to sync multiple catalogs. If you enable this, you'll be able to [filter catalogs](#catalogs-and-schemas).

Reviewer: This makes it seem
```suggestion
Toggle on to sync multiple catalogs. If you enable this, you'll be able to [specify which catalogs to sync](#catalogs-and-schemas).
```

otherwise it seems like _restricting_ the catalogs is the selling point, but it's the other way around

### Default catalog

> Required. You must specify a default catalog (so you don't have to deail with catalog qualification in native queries).

Reviewer: ```suggestion
Required. You must specify a default catalog (so you don't have to deal with catalog qualification in native queries).
```

You can't sync Databricks's legacy catalogs, however, including the `samples` or `hive_metastore` catalogs.

### Catalogs and schemas

> You can specify which catalogs schemas you want to sync and scan. Options are:

Reviewer: ```suggestion
You can specify which catalogs and schemas you want to sync and scan. Options are:
```
